app/controllers/api/auth_controller.rb

```diff
@@ -190,35 +202,51 @@ def update
     current_user.updated_with_user_agent = request.user_agent
     result = @user_manager.update(current_user, params)
     if result[:error]
-      render json: result, status: 401
+      render json: result, status: :unauthorized
     else
       render json: result
     end
   end
 
   def auth_params
-    if verify_mfa == false
+    unless verify_mfa
       # error responses are handled by the verify_mfa method
       return
     end
 
+    unless params[:email]
+      render json: {
+        error: {
+          message: 'Please provide an email address.',
+        },
+      }, status: :bad_request
+      return
+    end
+
     auth_params = @user_manager.auth_params(params[:email])
-    if !auth_params
+
+    unless auth_params
       render json: pseudo_auth_params(params[:email])
-    else
-      render json: @user_manager.auth_params(params[:email])
+      return
     end
+
+    render json: auth_params
   end
 
   def pseudo_auth_params(email)
     {
       identifier: email,
-      pw_cost: 110000,
+      pw_cost: 110_000,
       pw_nonce: Digest::SHA2.hexdigest(email + Rails.application.secrets.secret_key_base),
       version: '003',
```

Comment:
Should we change this to the latest version (`004`)? The `pseudo_auth_params` method is called when no `auth_params` exist for the specified email address. In this case, we want to use the latest version for new user registrations.

Reply:
Yup, makes sense.